### PR TITLE
expression: implement vectorized evaluation for `builtinRpadSig`

### PR DESCRIPTION
**Pull request by b41sh**
### What problem does this PR solve? 
Implement vectorized evaluation for `builtinRpadSig`.
[12106](https://github.com/pingcap/tidb/issues/12106)

### What is changed and how it works?
```
goos: darwin
goarch: amd64
pkg: github.com/pingcap/tidb/expression
BenchmarkVectorizedBuiltinStringFunc/builtinRpadSig-VecBuiltinFunc-4     5227  234686 ns/op   5616 B/op   434 allocs/op
BenchmarkVectorizedBuiltinStringFunc/builtinRpadSig-NonVecBuiltinFunc-4  4006  310491 ns/op  22080 B/op  1408 allocs/op
```

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: expression/builtin_string_vec_test.go [hunk 1]
```diff
@@ -87,10 +87,21 @@ var vecBuiltinStringCases = map[string][]vecExprBenchCase{
...
-	ast.Lpad:      {},
-	ast.Rpad:      {},
+	ast.LTrim: {},
+	ast.RTrim: {},
+	ast.Lpad:  {},
+	ast.Rpad: {
+		{
+			retEvalType:   types.ETString,
+			childrenTypes: []types.EvalType{types.ETString, types.ETInt, types.ETString},
+			geners:        []dataGenerator{&randLenStrGener{0, 20}, &rangeInt64Gener{0, 20}, &randLenStrGener{0, 10}},
```
Comment (qw4990) [inline expression/builtin_string_vec_test.go]: How about adding some cases to test this check this condition `uint64(targetLength)*uint64(mysql.MaxBytesOfCharacter) > b.maxAllowedPacket`?

Comment (b41sh) [reply expression/builtin_string_vec_test.go]: Ok

**Pull request merged**